### DISCUSSION TITLE
Can regular network requests be used instead of WebSocket?

### DISCUSSION THREAD

**Discussion by noontee0** [zauberzeug/nicegui#3953] [Ideas / Feature Requests]
I have discovered something called Mesop, which is a similar product, but it does not use websocket. Instead, it uses regular GET requests to achieve front-end and back-end communication. Is this method more advantageous than websocket Because in my opinion, websocket is a type of long-term connection that can cause users to idle web tags and occupy our server resources for a long time
Can this technology be integrated into Nicegui
Mesop's experience address: https://google.github.io/mesop/demo/ Here, you can take a look at its communication method

**Comment by rodja** [comment]
NiceGUI uses socket.io to implement its WebSocket connection, and it can be configured to use HTTP requests (polling) instead of WebSocket. The main benefits of WebSocket are:

1. **Bidirectional Communication**: WebSocket enables full-duplex communication, meaning both the client and server can send messages independently. This is especially useful for interactive applications that require real-time updates and feedback, like live dashboards or collaborative tools.
2. **Low Latency and Reduced Overhead**: Since WebSocket maintains an open connection, it avoids the overhead of repeated HTTP request headers, making it more efficient for applications with frequent data exchange. The persistent connection minimizes the time spent on re-establishing connections, thus reducing latency.
3. **Scalability for High-Volume Interactions**: WebSocket is ideal for applications with high user interaction rates, as it reduces server load compared to HTTP polling, which requires frequent requests that can overwhelm the server if scaled up.
4. **Efficient Resource Management**: Unlike long-polling or regular HTTP requests that can consume significant resources, WebSocket's persistent connection allows the server to handle idle connections more efficiently, reducing CPU and memory usage.

I think Mesop uses HTTP mainly to reduce complexity. I would not be surprised if they switch to websockets in a later version.

**Comment by mpechter** [comment]
If anyone is looking for a way to deploy NiceGUI on AWS, a great option is App Runner. However, App Runner doesn't support WebSockets. By forcing Socket.IO to use long polling instead, you can deploy there without issue:

```py
from nicegui import ui, core

core.app.config.socket_io_js_transports = ["polling"]
ui.run(proxy_headers=True, forwarded_allow_ips="*")
```

The proxy_headers and forwarded_allow_ips settings are needed because App Runner terminates TLS and forwards traffic as HTTP — without them, uvicorn doesn't know the client connected over HTTPS.
